oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/components/I18nTextInputField/I18nTextInputField.jsx

```diff
@@ -1,53 +1,46 @@
 import * as React from "react";
-import { LanguageSelectField, useSanitizeInput } from "@js/oarepo_ui";
-import { TextField, GroupField, FieldLabel } from "react-invenio-forms";
+import {
+  LanguageSelectField,
+  useSanitizeInput,
+  useFieldData,
+} from "@js/oarepo_ui";
+import { TextField, GroupField } from "react-invenio-forms";
 import PropTypes from "prop-types";
 import { useFormikContext, getIn } from "formik";
 
 export const I18nTextInputField = ({
   fieldPath,
-  label,
-  required,
   optimized,
-  labelIcon,
-  placeholder,
   lngFieldWidth,
   usedLanguages,
-  validTags,
   ...uiProps
 }) => {
   const { values, setFieldValue, setFieldTouched } = useFormikContext();
+
+  const { getFieldData } = useFieldData();
   const { sanitizeInput } = useSanitizeInput();
+  const lngFieldPath = `${fieldPath}.lang`;
+  const textFieldPath = `${fieldPath}.value`;
 
   return (
-    <GroupField fieldPath={fieldPath} optimized>
+    <GroupField fieldPath={fieldPath} optimized={optimized}>
       <LanguageSelectField
-        fieldPath={`${fieldPath}.lang`}
-        placeholder=""
-        required
+        fieldPath={lngFieldPath}
         width={lngFieldWidth}
         usedLanguages={usedLanguages}
+        {...getFieldData({ fieldPath: lngFieldPath, icon: "globe" })
```

Comment:
Shouldn't icon be just regular prop outside hook function?

Reply:
The icon is used by FieldLabel cmp (invenio) and for uniformity sake also for CompactLabel cmp (ours), which is why it is needed to return proper jsx